suse_migration_services/units/migrate.py

```diff
@@ -39,23 +40,47 @@ def main():
 
     try:
         log.info('Running migrate service')
-        bash_command = ' '.join(
-            [
-                'zypper', 'migration',
-                '--non-interactive',
-                '--gpg-auto-import-keys',
-                '--no-selfupdate',
-                '--auto-agree-with-licenses',
-                '--strict-errors-dist-migration',
-                '--replacefiles',
-                '--product', MigrationConfig().get_migration_product(),
-                '--root', root_path,
-                '&>>', Defaults.get_migration_log_file()
-            ]
-        )
-        Command.run(
-            ['bash', '-c', bash_command]
-        )
+        migration_config = MigrationConfig()
+        if migration_config.is_zypper_migration_plugin_requested():
```

Comment:
Reading this part of the code, it seems that this case is not the default one.

Reply:
sorry don't understand what you mean ?

Reply:
It's just nitpicking, it gives the impression that the use of zypper migration plugin is the special case.

Reply:
Oh that's what you mean. Well frankly speaking the migration plugin is a special case :) Usually people upgrade via ```zypper dup``` ;) If we want to change this I think also the logic of the option in the config file should change. I choose that way because the migration plugin has this clear name and is a plugin that one can use or not. ```zypper dup``` on the other hand is just an option to zypper and the long form is _distribution-upgrade_. I didn't find a nice name to express this mode in the config file and so I choose to go with "migration_plugin_requested y|n"

Currently I don't think we need to change this, but if you think it has a big potential for user confusion or misunderstanding we can of course reverse the logic ?

Reply:
I don't think we should change it either. It is clear that unless user provides another option, the zypper migration plugin is the default. Let's keep it like that and we will check with feedback in the future.

Reply:
ok, also remember there is #99 so we also document it and that should make the default clear.
Good then I'll merge it ;)

Thanks